classes/CustomText.php

```diff
@@ -39,11 +39,116 @@ class CustomText extends ObjectModel
 		'table' => 'info',
 		'primary' => 'id_info',
 		'multilang' => true,
+		'multilang_shop' => true,
 		'fields' => array(
-			'id_shop' =>			array('type' => self::TYPE_NOTHING, 'validate' => 'isUnsignedId'),
+			'id_info' =>	array('type' => self::TYPE_NOTHING, 'validate' => 'isUnsignedId'),
+			'id_shop' =>	array('type' => self::TYPE_NOTHING, 'validate' => 'isUnsignedId'),
 			// Lang fields
-			'text' =>				array('type' => self::TYPE_HTML, 'lang' => true, 'validate' => 'isCleanHtml', 'required' => true),
+			'text' =>		array('type' => self::TYPE_HTML, 'lang' => true, 'validate' => 'isCleanHtml', 'required' => true),
 		)
 	);
 
+	/**
+	 * Adding new custom text.
+	 * @param bool $autoDate
+	 * @param bool $nullValues
+	 * @return bool
+	 * @throws PrestaShopDatabaseException
+	 * @throws PrestaShopException
+	 */
+	public function add($autoDate = true, $nullValues = false)
+	{
+		$this->id = $this->id_info;
+
+		if (!$result = Db::getInstance()->insert($this->def['table'], $this->getFields(), $nullValues)) {
+			return false;
+		}
+
+		// Database insertion for multilingual fields related to the object
+		if (!empty($this->def['multilang'])) {
+			$fields = $this->getFieldsLang();
+			if ($fields && is_array($fields)) {
+				$asso = Shop::getAssoTable($this->def['table'].'_lang');
```

Comment:
Can you rename it to a meaningful variable? I can't get what information you retrieve, and why